packages/terra-disclosure-manager/src/DisclosureManager.jsx

```diff
@@ -407,9 +407,19 @@ class DisclosureManager extends React.Component {
    */
   generatePopFunction(key) {
     return () => {
-      let promiseRoot = Promise.resolve();
+      const { disclosureComponentKeys } = this.state;
+
+      if (disclosureComponentKeys[disclosureComponentKeys.length - 1] !== key) {
```

Comment:
With this change, can we remove the `disclosureComponentKeys.length` check throughout this file?

Reply:
I might be missing something, but I don't think so

Reply:
I'm not sure what I was thinking here... 🤷‍♀️